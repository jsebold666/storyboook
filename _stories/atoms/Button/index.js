import React from 'react';
import { text, selectV2 as select, boolean, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import Button from '../../../components/atoms/Button';

const options = {
  small: 'small',
  medium: 'medium',
  large: 'large'
};

const component = () => (
  <Button
    isLoading={boolean('isLoading', false)}
    onClick={action('button_clicked')}
    showIcon={boolean('showIcon', false)}
    className={text('class', '')}
    disabled={boolean('disabled', false)}
    size={select('Size', options)}>
    Btn Text
  </Button>
);

export default [readme, component];
