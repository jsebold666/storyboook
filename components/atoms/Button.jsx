import React from 'react';
import PropTypes from 'prop-types';
import ArrowForward from '@material-ui/icons/ArrowForward';
import LoadingIcon from './LoadingIcon';
import styled from 'styled-components';
import shared from '../consts/shared';

const Btn = styled.button`
  &:not(.btn-loading),
  &:not(.btn-loading) * {
    cursor: pointer;
  }

  &.btn {
    min-width: 200px;
    width: 100%;

    border: 0;

    outline: none;
    background-color: ${shared['tealish']};
    color: ${shared['white']};

    &.small {
      height: 48px;
      padding: 12px 24px 14px 24px;
      border-radius: 4px;
    }

    &.medium {
      height: 80px;
      padding: 24px 16px 24px 24px;
    }

    &.large {
      height: 100px;
      padding: 24px 16px 24px 24px;
    }

    &:hover:not(.btn-default):not(.btn-loading):not([disabled]) {
      background-color: #43e4e4;
    }
    &:active:not(.btn-default):not(.btn-loading):not([disabled]) {
      background-color: #2fbcbc;
    }
  }

  &.btn-default {
    color: ${shared['periwinkle']};
    background-color: ${shared['white']};
  }

  &:disabled {
    background-color: ${shared['cloudy-blue']};
    &,
    & * {
      cursor: not-allowed;
    }
  }

  & .btn-inner__container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    & label {
      font-size: 16px;
      font-weight: 500;
      font-style: normal;
      font-stretch: normal;
      line-height: 1.5;
      letter-spacing: normal;
      margin-right: 32px;
      flex: 1;
      text-align: left;
    }
  }

  .btn-bottom {
    position: absolute;
    bottom: 0;
  }
`;

class Button extends React.Component {
  constructor() {
    super();

    this.getIcon = this.getIcon.bind(this);
    this.onClickHandler = this.onClickHandler.bind(this);
  }

  onClickHandler(ev) {
    const { onClick, isLoading } = this.props;

    if (onClick && !isLoading) {
      onClick(ev);
    }
  }

  getIcon() {
    if (this.props.isLoading) {
      const { className } = this.props;
      const isWhiteBackground = className && className.includes('btn-default');
      const color = isWhiteBackground ? shared['periwinkle'] : shared['white'];
      return <LoadingIcon color={color} />;
    } else if (this.props.showIcon) {
      return <ArrowForward />;
    }
  }

  render() {
    const { className, disabled, isLoading, type, size } = this.props;
    const klass = `btn ${className || ''} ${isLoading ? 'btn-loading' : ''} ${size || 'medium'}`;
    return (
      <Btn type={type} className={klass} disabled={disabled} onClick={this.onClickHandler}>
        <div className="btn-inner__container">
          <label>{this.props.children}</label>
          {this.getIcon()}
        </div>
      </Btn>
    );
  }
}

Button.defaultProps = {
  isLoading: false,
  showIcon: true,
  disabled: false,
  type: 'button'
};

Button.propTypes = {
  type: PropTypes.string,
  isLoading: PropTypes.bool,
  showIcon: PropTypes.bool,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};

export default Button;
