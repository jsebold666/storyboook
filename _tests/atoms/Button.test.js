/* globals mount */
import React from 'react';
import Button from '../../components/atoms/Button';

const defaultProps = {
  group: false,
  size: 'sm',
  onClick: () => {},
  text: 'This is text',
  context: 'danger',
  className: 'foo-class',
  style: {
    width: '1000px'
  },
  onBlur: () => {}
};

describe('Expect <Button>', () => {
  it('to render', () => {
    const wrapper = mount(<Button {...defaultProps}>My Cool Button</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it('to renders as a group button', () => {
    const props = {
      ...defaultProps,
      group: true
    };
    const wrapper = mount(<Button {...props}>My Cool Button</Button>);
    expect(wrapper).toMatchSnapshot();
  });
});
