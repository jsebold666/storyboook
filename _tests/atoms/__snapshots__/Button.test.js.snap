// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Button> to render 1`] = `
<Button
  className="foo-class"
  context="danger"
  disabled={false}
  group={false}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  showIcon={true}
  size="sm"
  style={
    Object {
      "width": "1000px",
    }
  }
  text="This is text"
  type="button"
>
  <Button__Btn
    className="btn foo-class  sm"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <StyledComponent
      className="btn foo-class  sm"
      disabled={false}
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Button__Btn-nzgghs-0",
            "isStatic": true,
            "lastClassName": "hEKkqr",
            "rules": Array [
              "
  &:not(.btn-loading),
  &:not(.btn-loading) * {
    cursor: pointer;
  }

  &.btn {
    min-width: 200px;
    width: 100%;

    border: 0;

    outline: none;
    background-color: ",
              "#33cccc",
              ";
    color: ",
              "#fff",
              ";

    &.small {
      height: 48px;
      padding: 12px 24px 14px 24px;
      border-radius: 4px;
    }

    &.medium {
      height: 80px;
      padding: 24px 16px 24px 24px;
    }

    &.large {
      height: 100px;
      padding: 24px 16px 24px 24px;
    }

    &:hover:not(.btn-default):not(.btn-loading):not([disabled]) {
      background-color: #43e4e4;
    }
    &:active:not(.btn-default):not(.btn-loading):not([disabled]) {
      background-color: #2fbcbc;
    }
  }

  &.btn-default {
    color: ",
              "#9966ff",
              ";
    background-color: ",
              "#fff",
              ";
  }

  &:disabled {
    background-color: ",
              "#cfdae1",
              ";
    &,
    & * {
      cursor: not-allowed;
    }
  }

  & .btn-inner__container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    & label {
      font-size: 16px;
      font-weight: 500;
      font-style: normal;
      font-stretch: normal;
      line-height: 1.5;
      letter-spacing: normal;
      margin-right: 32px;
      flex: 1;
      text-align: left;
    }
  }

  .btn-bottom {
    position: absolute;
    bottom: 0;
  }
",
            ],
          },
          "displayName": "Button__Btn",
          "render": [Function],
          "styledComponentId": "Button__Btn-nzgghs-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      type="button"
    >
      <button
        className="btn foo-class  sm Button__Btn-nzgghs-0 hEKkqr"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <div
          className="btn-inner__container"
        >
          <label>
            My Cool Button
          </label>
          <pure(ArrowForwardIcon)>
            <ArrowForwardIcon>
              <WithStyles(SvgIcon)>
                <SvgIcon
                  classes={
                    Object {
                      "colorAction": "MuiSvgIcon-colorAction-4",
                      "colorDisabled": "MuiSvgIcon-colorDisabled-6",
                      "colorError": "MuiSvgIcon-colorError-5",
                      "colorPrimary": "MuiSvgIcon-colorPrimary-2",
                      "colorSecondary": "MuiSvgIcon-colorSecondary-3",
                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit-7",
                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge-9",
                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall-8",
                      "root": "MuiSvgIcon-root-1",
                    }
                  }
                  color="inherit"
                  component="svg"
                  fontSize="default"
                  viewBox="0 0 24 24"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-1"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </SvgIcon>
              </WithStyles(SvgIcon)>
            </ArrowForwardIcon>
          </pure(ArrowForwardIcon)>
        </div>
      </button>
    </StyledComponent>
  </Button__Btn>
</Button>
`;

exports[`Expect <Button> to renders as a group button 1`] = `
<Button
  className="foo-class"
  context="danger"
  disabled={false}
  group={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  showIcon={true}
  size="sm"
  style={
    Object {
      "width": "1000px",
    }
  }
  text="This is text"
  type="button"
>
  <Button__Btn
    className="btn foo-class  sm"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <StyledComponent
      className="btn foo-class  sm"
      disabled={false}
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Button__Btn-nzgghs-0",
            "isStatic": true,
            "lastClassName": "hEKkqr",
            "rules": Array [
              "
  &:not(.btn-loading),
  &:not(.btn-loading) * {
    cursor: pointer;
  }

  &.btn {
    min-width: 200px;
    width: 100%;

    border: 0;

    outline: none;
    background-color: ",
              "#33cccc",
              ";
    color: ",
              "#fff",
              ";

    &.small {
      height: 48px;
      padding: 12px 24px 14px 24px;
      border-radius: 4px;
    }

    &.medium {
      height: 80px;
      padding: 24px 16px 24px 24px;
    }

    &.large {
      height: 100px;
      padding: 24px 16px 24px 24px;
    }

    &:hover:not(.btn-default):not(.btn-loading):not([disabled]) {
      background-color: #43e4e4;
    }
    &:active:not(.btn-default):not(.btn-loading):not([disabled]) {
      background-color: #2fbcbc;
    }
  }

  &.btn-default {
    color: ",
              "#9966ff",
              ";
    background-color: ",
              "#fff",
              ";
  }

  &:disabled {
    background-color: ",
              "#cfdae1",
              ";
    &,
    & * {
      cursor: not-allowed;
    }
  }

  & .btn-inner__container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    & label {
      font-size: 16px;
      font-weight: 500;
      font-style: normal;
      font-stretch: normal;
      line-height: 1.5;
      letter-spacing: normal;
      margin-right: 32px;
      flex: 1;
      text-align: left;
    }
  }

  .btn-bottom {
    position: absolute;
    bottom: 0;
  }
",
            ],
          },
          "displayName": "Button__Btn",
          "render": [Function],
          "styledComponentId": "Button__Btn-nzgghs-0",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      type="button"
    >
      <button
        className="btn foo-class  sm Button__Btn-nzgghs-0 hEKkqr"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <div
          className="btn-inner__container"
        >
          <label>
            My Cool Button
          </label>
          <pure(ArrowForwardIcon)>
            <ArrowForwardIcon>
              <WithStyles(SvgIcon)>
                <SvgIcon
                  classes={
                    Object {
                      "colorAction": "MuiSvgIcon-colorAction-4",
                      "colorDisabled": "MuiSvgIcon-colorDisabled-6",
                      "colorError": "MuiSvgIcon-colorError-5",
                      "colorPrimary": "MuiSvgIcon-colorPrimary-2",
                      "colorSecondary": "MuiSvgIcon-colorSecondary-3",
                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit-7",
                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge-9",
                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall-8",
                      "root": "MuiSvgIcon-root-1",
                    }
                  }
                  color="inherit"
                  component="svg"
                  fontSize="default"
                  viewBox="0 0 24 24"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-1"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </SvgIcon>
              </WithStyles(SvgIcon)>
            </ArrowForwardIcon>
          </pure(ArrowForwardIcon)>
        </div>
      </button>
    </StyledComponent>
  </Button__Btn>
</Button>
`;
